### PR TITLE
Added route-manager create method

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | none
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR add the route-manager create and automatic route creation.

#### Todo

- [x] collision detection
- [x] tests

### CODE REVIEW

File: Generator/RouteGenerator.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+     * @var TokenProviderInterface
+     */
+    private $tokenProvider;
+
+    /**
+     * @var SlugifierInterface
+     */
+    private $slugifier;
+
+    public function __construct(TokenProviderInterface $tokenProvider, SlugifierInterface $slugifier)
```
Comment (chirimoya) [inline Generator/RouteGenerator.php]: phpdoc

File: Manager/ConflictResolverInterface.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\Manager;
+
+use Sulu\Bundle\RouteBundle\Model\RouteInterface;
+
+interface ConflictResolverInterface
```
Comment (chirimoya) [inline Manager/ConflictResolverInterface.php]: phpdoc

File: Manager/RouteAlreadyCreatedException.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+ * This exception indicates that a entity already has a route.
+ */
+class RouteAlreadyCreatedException extends \Exception
+{
+    /**
+     * @var RoutableInterface
+     */
+    private $entity;
+
+    public function __construct(RoutableInterface $entity)
```
Comment (chirimoya) [inline Manager/RouteAlreadyCreatedException.php]: phpdoc

File: Manager/RouteManager.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+     * @var RouteRepositoryInterface
+     */
+    private $routeRepository;
+
+    /**
+     * @var ConflictResolverInterface
+     */
+    private $conflictResolver;
+
+    public function __construct(
```
Comment (chirimoya) [inline Manager/RouteManager.php]: phpdoc

File: Generator/RouteGenerator.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public function generate($entity, $routeSchema)
+    {
+        $tokens = [];
+        preg_match_all('/{(.*?)}/', $routeSchema, $matches);
+        $tokenNames = $matches[1];
+
+        foreach ($tokenNames as $index => $name) {
+            $tokenName = '{'.$name.'}';
+            $tokenValue = $this->tokenProvider->provide($entity, $name);
```
Comment (chirimoya) [inline Generator/RouteGenerator.php]: How can we handle translations?

**Comment by chirimoya** [comment]
ACK

File: Generator/RouteGenerator.php (hunk 5 above)
Comment (wachterjohannes) [inline Generator/RouteGenerator.php]: I will introduce a symfony expression token provider which will be able to handle translations. But that will go into a own PR.

Comment (wachterjohannes) [inline Generator/RouteGenerator.php]: added in the next pr https://github.com/sulu/SuluRouteBundle/pull/4/files#r59364214

**Pull request merged**